docs(security): add info about CorsOptionsDelegate (#1729)
nestjs/docs.nestjs.com: content/security/cors.md

await app.listen(3000);
```

The `enableCors()` method takes an optional configuration object argument. The available properties of this object are described in the official [CORS](https://github.com/expressjs/cors#configuration-options) documentation. Another way is to pass a callback function called [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) that lets you define the configuration ovject asynchronously based on the request (on the fly).

Alternatively, enable CORS via the `create()` method's options object. Set the `cors` property to `true` to enable CORS with default settings. 
Or, pass a [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value to customize its behavior.

```typescript
const app = await NestFactory.create(AppModule, { cors: true });
await app.listen(3000);
```

> Above method only applies to REST endpoints. 
> To enable CORS in GraphQL, set `cors` property to `true` or pass [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronouslyhttps://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value when you import GraphQL module.

Reviewer: ```suggestion
Above method only applies to REST endpoints. 

To enable CORS in GraphQL, set `cors` property to `true` or pass [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value when you import GraphQL module.
```

await app.listen(3000);
```

> The `enableCors()` method takes an optional configuration object argument. The available properties of this object are described in the official [CORS](https://github.com/expressjs/cors#configuration-options) documentation. Another way is to pass a callback function called [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) that lets you define the configuration ovject asynchronously based on the request (on the fly).

Reviewer: ```suggestion
The `enableCors()` method takes an optional configuration object argument. The available properties of this object are described in the official [CORS](https://github.com/expressjs/cors#configuration-options) documentation. Another way is to pass a [callback function](https://github.com/expressjs/cors#configuring-cors-asynchronously) that lets you define the configuration object asynchronously based on the request (on the fly).
```

> The `enableCors()` method takes an optional configuration object argument. The available properties of this object are described in the official [CORS](https://github.com/expressjs/cors#configuration-options) documentation.  
> Another way is to pass a function called [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronouslyhttps://github.com/expressjs/cors#configuring-cors-asynchronously) that prepares the configuration object asynchronously based on the request.

Reviewer: ```suggestion
The `enableCors()` method takes an optional configuration object argument. The available properties of this object are described in the official [CORS](https://github.com/expressjs/cors#configuration-options) documentation. Alternatively, pass a callback function called [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronouslyhttps://github.com/expressjs/cors#configuring-cors-asynchronously) that prepares the configuration object asynchronously based on the incoming request.
```

> The `enableCors()` method takes an optional configuration object argument. The available properties of this object are described in the official [CORS](https://github.com/expressjs/cors#configuration-options) documentation.  
> Another way is to pass a function called [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronouslyhttps://github.com/expressjs/cors#configuring-cors-asynchronously) that prepares the configuration object asynchronously based on the request.

Reviewer: ```suggestion
The `enableCors()` method takes an optional configuration object argument. The available properties of this object are described in the official [CORS](https://github.com/expressjs/cors#configuration-options) documentation. Another way is to pass a callback function called [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) that lets you define the configuration ovject asynchronously based on the request (on the fly).
```

Alternatively, enable CORS via the `create()` method's options object. Set the `cors` property to `true` to enable CORS with default settings. 

> Or, pass a [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value to customize its behavior.

Reviewer: ```suggestion
Or, pass a [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [callback function](https://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value to customize its behavior.
```

> Alternatively, enable CORS via the `create()` method's options object. Set the `cors` property to `true` to enable CORS with default settings. 
> Or, pass a [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronouslyhttps://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value to customize its behavior.

Reviewer: ```suggestion
Alternatively, enable CORS via the `create()` method's options object. Set the `cors` property to `true` to enable CORS with default settings. 
Or, pass a [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value to customize its behavior.
```

```typescript
const app = await NestFactory.create(AppModule, { cors: true });
await app.listen(3000);
```

Above method only applies to REST endpoints. 

> To enable CORS in GraphQL, set `cors` property to `true` or pass [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or [CorsOptionsDelegate](https://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value when you import GraphQL module.

Reviewer: ```suggestion
To enable CORS in GraphQL, set `cors` property to `true` or pass [CORS configuration object](https://github.com/expressjs/cors#configuration-options) or a [callback function](https://github.com/expressjs/cors#configuring-cors-asynchronously) as the `cors` property value when you import GraphQL module.
```

> > warning **Warning** CorsOptionsDelegate is not working with apollo-server-fastify yet.

Reviewer: ```suggestion
> warning **Warning** `CorsOptionsDelegate` solution is not working with the `apollo-server-fastify` package yet.
```
